src/core/api/udp_api.cpp

```diff
@@ -44,7 +44,8 @@ otMessage *otUdpNewMessage(otInstance *aInstance, const otMessageSettings *aSett
 
 otError otUdpOpen(otInstance *aInstance, otUdpSocket *aSocket, otUdpReceive aCallback, void *aContext)
 {
-    return AsCoreType(aInstance).Get<Ip6::Udp>().Open(AsCoreType(aSocket), Ip6::kNetifThreadHost, aCallback, aContext);
+    return AsCoreType(aInstance).Get<Ip6::Udp>().Open(AsCoreType(aSocket), Ip6::kNetifThreadInternal, aCallback,
+                                                      aContext);
```

Comment:
I think this may be correct and intended.
* It is the only place where `kNetifThreadHost` is used.
* This is the public OT API `otUpdOpen` for other software entities (not the OpenThread stack itself) that want to create a UDP socket. The user of this socket may, therefore, expect it to be available on the platform-side socket as well.
* The `kNetifThreadInternal` is mainly intended for sockets created by the OT stack itself which we know for sure are not meant to be exposed to the UDP platform.